### PR TITLE
Allow admins to resend email verification to users

### LINKED ISSUE #1210: Allow admins to request lexbox send verification email for user
**Describe the feature**
Sometimes a user can't find a verification email an admin would like to help them by requesting lexbox send them a verification email.

**Who is this feature for?**
Admins

**Pages affected**
Admin page, menu item in user dropdown

### PR DESCRIPTION
**Pull request by rmunn**
Fix #1210.

User modal now has an admin-only button to resend verification email to any users who aren't verified. Button only shows up if an admin is logged in *and* the user is unverified.

screenshot

### CODE REVIEW

**Comment by rmunn** [comment]
Just realized that the way I implemented this, the button will only do something if the user modal was accessed from the admin dashboard. I did that because I thought "We shouldn't put admin-dashboard only code into the user modal" so I put the "resend email" call in the admin dashboard page, rather than in the modal. But really, we want the modal to work the same way no matter what page it's on. So I'm going to move the "resend email" code into the modal itself, so that it will also work if an admin gets to that user modal from clicking on an org member on the org page, or a project member on the project page.

**Review by hahn-kev — COMMENTED** [review]
it looks good to me, though I'm not sure I like the idea of putting actions in the user modal, if only because there are no other ones there, and it won't be obvious to go there to send the email, I noted on the issue that it might make sense to be a menu item on the admin dashboard. @myieye what do you think? Feel free to override my suggestion.

**Comment by myieye** [comment]
> it looks good to me, though I'm not sure I like the idea of putting actions in the user modal, if only because there are no other ones there, and it won't be obvious to go there to send the email, I noted on the issue that it might make sense to be a menu item on the admin dashboard. @myieye what do you think? Feel free to override my suggestion.

Tricky 🤔...

TLDR;
I like where @rmunn has it now.


Long version:
Putting things in the User details modal has the added benefit that it's available on the admin dashboard, project page and org page.

Ultimately, I like the idea of merging the edit user and user details modals. (e.g. Having an "Edit" button in the User details modal that replaces the modal contents with the Edit form.) It's strange that you have to close the user details modal and open a different one in order to edit the user. Merging them would have the added benefit of also being able to edit a user in the context of a project or org.

**I'm not proposing we make that change in this PR! 😉**

But that change would result in the user details modal being "one user modal to rule them all". So, I think that's where the "Resend verification email" button belongs.

**Review by myieye — CHANGES_REQUESTED** [review]
I like this so far 👍 
But, there are a few changes I'd like to see 🙂

File: frontend/src/lib/components/Users/UserModal.svelte [hunk 1]
```diff
@@ -27,6 +32,14 @@
     user = _user;
     await userDetailsModal.openModal(true, true);
   }
+
+  const { notifySuccess } = useNotifications();
+
+  // eslint-disable-next-line func-style
```
[the remark this answers is not in the document]
Comment (rmunn) [reply frontend/src/lib/components/Users/UserModal.svelte:38]: eslint doesn't think so. As far as I can tell, because there's a `function foo() { }` style definition elsewhere in the component, it thinks every function definition needs to be in that style and rejects `bar = (params) => { doSomething(); }`. But since Svelte wants an `export let` here, I can't switch to the `function bar() { }` style here as the Svelte compiler treats that differently.

Basically eslint is fighting with the Svelte compiler, and this was the best way to get eslint to shut up.

File: frontend/src/lib/components/Users/UserModal.svelte [hunk 2]
```diff
@@ -50,6 +63,14 @@
                 data-tip={$t('admin_dashboard.email_not_verified')}>
                 <span class="i-mdi-help-circle-outline" />
               </span>
+              <AdminContent>
+                <IconButton
+                  size="btn-sm"
+                  icon="i-mdi-email"
+                  title={$t('admin_dashboard.resend_verification_email')}
```
[the remark this answers is not in the document]
Comment (rmunn) [reply frontend/src/lib/components/Users/UserModal.svelte:70]: Right, I tried `data-tip` on the button and it didn't work. I'll wrap it with a div as suggested.

File: frontend/src/lib/components/Users/UserModal.svelte [hunk 3]
```diff
@@ -50,6 +63,14 @@
                 data-tip={$t('admin_dashboard.email_not_verified')}>
                 <span class="i-mdi-help-circle-outline" />
               </span>
+              <AdminContent>
+                <IconButton
+                  size="btn-sm"
+                  icon="i-mdi-email"
```
[the remark this answers is not in the document]
Comment (rmunn) [reply frontend/src/lib/components/Users/UserModal.svelte:69]: Which one is the one with the circle arrows? I didn't see that one in https://materialdesignicons.com/cdn/1.6.50-dev/. That looks like the best choice, but I'm not finding its name easily.

File: frontend/src/lib/components/Users/UserModal.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (rmunn) [reply frontend/src/lib/components/Users/UserModal.svelte:38]: Oh wait, that was the previous version of the code where I exposed a function prop. Never mind, that comment can go now.

[the remark this answers is not in the document]
Comment (rmunn) [reply frontend/src/lib/components/Users/UserModal.svelte]: Removed in commit 41391908.

File: frontend/src/lib/components/Users/UserModal.svelte (hunk 2 above)
[the remark this answers is not in the document]
Comment (rmunn) [reply frontend/src/lib/components/Users/UserModal.svelte]: Done in commit 41391908.

File: frontend/src/lib/components/Users/UserModal.svelte (hunk 3 above)
[the remark this answers is not in the document]
Comment (rmunn) [reply frontend/src/lib/components/Users/UserModal.svelte]: Thanks for the help on Slack. Changed icon to `email-sync` in commit ba275607.

[the remark this answers is not in the document]
Comment (myieye) [reply frontend/src/lib/components/Users/UserModal.svelte]: For future reference, I'm thinking this looks like the best place to search for icons: https://pictogrammers.com/library/mdi/

It shows what [version](https://www.npmjs.com/package/@mdi/svg?activeTab=versions) icons were introduced as well as aliases of the icon, so it's easier to find the "main name" of an icon.

**Review by myieye — APPROVED** [review]
I pushed a few changes:
1) The notification was hidden behind the modal, so I put notifications inside our Modal component and added a demo to our sandbox page that seems to look fine. I believe the notifications are now in 2 places in the dom if they're in a modal, but it's visually impossible to see that, so I'm happy.
image

2) I made the button our primary color. It just didn't look enough like a button for my taste 😄 
image

3) I added a loading state to the button (partially because it can take a while to send an email)
image
